## 5.6 Step VI - EO Payload Sizing relations 


Goal of this step is to provide EO payload sizing rough order of magnitude (RoM) sizing relations to addressed two main questions:
- Can we "see/distinguish" the phenomenon/thing of interest? Which is answered by the EO resolution metrics
- Can we download the data on time? Which is to be answered with the input of orbital simulation yielding contact times with ground segment (ground stations) and the data generation as a function or orbit over area of interest (AoI) at nadir times a data rate (part two of this code)


![PhD IPO Diagrams for process.png](Figs/PhD%20IPO%20Diagrams%20for%20process.png)


### Earth Observation Payload Sizing relations (Context i.e. what is feasible) 
Based on [Valenzuela and Reyes 2019] the metrics for resolution $R$ where implemented in the following phase.

This is done to size the payload spatial resolution based on analogy to other missions and provide initial values for the mission analysis by constraining the "physically" feasible values (i.e. the finer spatial resolution achievable given the boundary/constrains of the CubeSat).

Note that in the discussion with small satellite developers the constrains of te CubeSat standard in terms of its dimensions where mentions as rationale for moving to microsatellites.

#### INPUT: Resolution definition

- $ p $ IS DETECTOR LEVEL PITCH
- $ f $ IS FOCAL LENGTH
- $ LAMBD $ IS $ \lambda $, AVERAGE WAVELENGTH
- $ H $ IS ALTITUDE 

Values are taken from TABLE 3-1 Lopez & dos Santos 2020 WETE

In [1]:
import numpy as np
import pandas as pd
## INITIALIZATION OF VARIABLES
# Defines vectors with all the values on Lopez & dos Santos 2020 to calculate the different spatial resolution metrics by Valenzuela & Reyes 2019

p=np.array([7.4e-6, 5.5e-6, 5.5e-6, 5.5e-6])
f=np.array([3.2, 0.9, 0.9, 0.3])
lambd=np.array([5.5e-7, 5.5e-7, 5.5e-7, 5.5e-7])
H=np.array([6.2e5, 6e5, 4e5, 4e5])
D=np.array([0.2, 0.09, 0.09, 0.09])

#### PROCESS: Earth observation resolution relationships 
Relationships adapted from [Valenzuela & Reyes 2019]
- Ground Sample Distance ($R_{GSD}$)
- Raleigh Diffraction Limit ($R_{RDL}$)
- Ground Spot Size ($R_{GSS}$)
- Optical factor ($Q$)



$$
R_{GSD}= p\frac{H}{f} [m]\\

R_{RDL}=1.22\frac{\lambda H}{D} [m]\\
 
R_{GSS}= \lambda\frac{H}{D} [m]\\

Q= \lambda\frac{f}{Dp} [-]
$$

In [2]:
## EQUATIONS

R_GSD= p*H/f # [m] Ground Sample Distance

R_RDL=1.22*lambd*H/D # Raleigh Diffraction Limit [m]

R_GSS= lambd*H/D # Ground Spot Size [m]

Q= lambd*f/(D*p) # Quality Factor [-]

In [3]:
## Generate OUTPUT summary TABLE
## Example values used in publication by Lopez-Telgie & dos Santos at WETE
data = {
    'Satellite': ['SSOT_SSO', '3U_Dove_SSO', '3U_Dove_ISS', '1U_Cubesat_ISS'], #Satellite names
    'Pixel pitch, p [m]': ['{:.2e}'.format(val) for val in p], #pixel in detector physical dimension (size) 
    'Satellite Altitude, H [m]': ['{:.2e}'.format(val) for val in H], 
    'Instrumental Focal Length, f [m]': f, 
    'Ground Sample Distance, R_GSD [m]': ['{:.3}'.format(val) for val in R_GSD],
    'Average Wavelength, $\lambda$ [m]': ['{:.2e}'.format(val) for val in lambd], 
    'Diameter, D [m]': D,
    'Rayleigh Diffraction Limit, R_RDL [m]': ['{:.3}'.format(val) for val in R_RDL],
    'Ground Spot Size, R_GSS [m]':['{:.3}'.format(val) for val in R_GSS],
    'Optical Factor,  Q [-]': ['{:.3}'.format(val) for val in Q]
}

df=pd.DataFrame(data)
display(df)

Satellite Pixel pitch, p [m] Satellite Altitude, H [m]  \
0        SSOT_SSO           7.40e-06                  6.20e+05   
1     3U_Dove_SSO           5.50e-06                  6.00e+05   
2     3U_Dove_ISS           5.50e-06                  4.00e+05   
3  1U_Cubesat_ISS           5.50e-06                  4.00e+05   

   Instrumental Focal Length, f [m] Ground Sample Distance, R_GSD [m]  \
0                               3.2                              1.43   
1                               0.9                              3.67   
2                               0.9                              2.44   
3                               0.3                              7.33   

  Average Wavelength, $\lambda$ [m]  Diameter, D [m]  \
0                          5.50e-07             0.20   
1                          5.50e-07             0.09   
2                          5.50e-07             0.09   
3                          5.50e-07             0.09   

  Rayleigh Diffraction Limit, R_RDL [m] Ground Spot Size, R_GSS [m]  \
0                                  2.08                        1.71   
1                                  4.47                        3.67   
2                                  2.98                        2.44   
3                                  2.98                        2.44   

  Optical Factor,  Q [-]  
0                   1.19  
1                    1.0  
2                    1.0  
3                  0.333

Note that DOVE are 3U Cubesat, constrained by the CDS to $10\times 10 \times 30 cm^3$ which ends in a maximum feasible aperture of 9 cm leaving some space for the structure, and the focal length on $~3\times$ the z-axis through the use of mirrors. While SSOT corresponds to a 140 kg small satellite deployed in 2011.  

## Sizing spatial resolution for orbital estimating data rates of payload

As said, previously the previous block worked as a context for understanding the capabilities as well as upper and lower limits of the constants being used in this problem. Specifically, it was probed that a GSD can be as low as 3.67 [m] however previous iterations of this problem showed a GSD as low as that is far demanding for the link budget. 

In this section, several values of spatial resolution in terms of $R_{GSD}$ and orbit altitudes are defined in order to iterate and define the most suitable scenario to accomplish this mission. 

#### Swath width discussion

Figure XX provides and overview of Planet sensors characteristics that are relevant for the sizing by analogy of our Payload.

![planetscopes sensors.jpg](Figs/planetscopes%20sensores.jpg)

Planet source: https://www.mdpi.com/2072-4292/13/19/3930

In [ ]:
## Scenario dependant parameters

R_GSD= 25 #[m] Spatial resolution

H= 4.5e5 #[m] Orbital altitude

S_w=np.array([19000, 22000, 25000, 30000, 35000, 45000]) ## [m] Estimated from equivalent CubeSat for fire monitoring mission [Azami et al., 2022]

The data rate from the payload (i.e generate: $_gen$) from the image is estimated from: 

$$ DR_{gen}= \frac{S_w V_g}{R_{GSD}}b \;\left[ \frac{bit}{s} \right] $$

Where
- $S_w$: Swath width [m] set as 25 [km] from [Azami et al., 2022]
- $b$ : bit rate [bit/s]. Set as 12 [bit/s] from KITSUNE forest monitoring CubeSat mission [Azami et al., 2022]
- $R_{GSD}$: Ground Sample Distance resolution [m], estimated previously based on the relationships by [Valenzuela & Reyes 2019].

To loosen, i.e. make the $R_{GSD} 25 [m]$ for FireSat scenario. Required resolution is 50 m  **(Requirement 34.1.2 Resolution: 50 meter resolution)**

- $V_g$: Ground speed [m/s], estimated as follows [New SMAD - Appendix C]:


$$ V_{g}=R_E \sqrt{\frac{\mu_E}{a^3}} [\frac{m}{s}]$$

Where:
- $\mu_E=3.986004356e14 [m^3/s^2]$ Earth Gravitational Constant  [New SMAD - Spaceflight constants]
- $a=R_E+H$ semi major axis [m]
- $R_{E}=6378136.6 [m]$ Earth equatorial radius [New SMAD - Spaceflight constants]
- $H$ orbital altitude (height) [m]. From the previous cases defined in the INPUT H vector

In [ ]:
## SSO a 600 [km],  GSD 25 [m] y S_w 25 [km]-> R_GSD[9], H[1],  S[2]

## Payload Data Rate sizing relations 
mu_E= 3.986004356e14 ## [m^3/s^2] [New SMAD - Spaceflight constants]
R_E=6378136.6 ## [m]$ Earth equatorial radius [m] [New SMAD - Spaceflight constants]

V_g=R_E*np.sqrt(mu_E/(R_E+H)**3) 
print('Ground speed velocity:', V_g/1000, ' [km/s]')

#V_g= np.sqrt(mu*(1/(R_E+H))) ## [m/s]

b=12 # [bit]

D_rategenerated= S_w[2]* V_g*b/(R_GSD) #[bit/s]
print('Payload data rate:', D_rategenerated / 1000000, ' [Mbit/s]')